include/superior_mysqlpp/extras/prepared_statement_utils.hpp

```diff
@@ -0,0 +1,135 @@
+        }
+
+        /**
+         * @brief Makes prepared statement from function's signature
+         * @param connection Connection handle into database
+         * @param query Query for prepared statement
+         */
+        template <typename CallableClass, typename... Args>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, void (CallableClass::*)(Args...) const) {
+            return connection.makePreparedStatement<ResultBindings<std::remove_cv_t<std::remove_reference_t<Args>>...>>(query);
+        }
+
+        /**
+         * @brief Makes prepared statement from function's signature
+         * @param connection Connection handle into database
+         * @param query Query for prepared statement
+         */
+        template <typename CallableClass, typename... Args>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, void (CallableClass::*)(Args...)) {
+            return connection.makePreparedStatement<ResultBindings<std::remove_cv_t<std::remove_reference_t<Args>>...>>(query);
+        }
+
+        /**
+         * @brief Makes prepared statement from lambda's signature
+         * @param connection Connection handle into database
+         * @param query Query for prepared statement
+         */
+        template <typename CallableClass>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, CallableClass&&) {
+            // This is actually only callable argument type deductor. We don't care about callable type, so we can decay Callable object
+            return generatePreparedStatementImpl(connection, query, &std::decay_t<CallableClass>::operator());
+        }
+
+        /**
+         * @brief Makes prepared statement from plain old C function's signature
+         * @param connection Connection handle into database
+         * @param query Query for prepared statement
+         */
+        template <typename... Args>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, void (*)(Args...)) {
+            return connection.makePreparedStatement<ResultBindings<std::remove_cv_t<std::remove_reference_t<Args>>...>>(query);
+        }
+    }
+
+    /**
+     * @brief Executes query, reads input data and passes them to callback function
+     * @param ps Prepared statement object
+     * @param processingFunction function to be invoked on every row
+     */
+    template<typename PreparedStatementType, typename Callable>
+    void psReadQuery(PreparedStatementType &&ps, Callable &&processingFunction)
+    {
+        ps.execute();
+        while(ps.fetch())
+        {
+            invokeViaTuple(processingFunction, ps.getResult());
+        }
+    }
+
+    /**
+     * @brief Builds prepared statement from string and invokes psReadQuery(ps, processingFunction)
```

Comment:
As above, please describe that result columns must correspond to processingFunction arguments